### PR TITLE
The big integration

### PR DESCRIPTION
**Pull request by janlauber**
solves the following issue(s):
- #114 
- #98 
- #99 
- #111
- #53 
- #52 
- #51

### CODE REVIEW

**Comment by janlauber** [comment]
can't continuing working on the integrations, bc I cannot run the backend on my machine.....
will continue after @Andy-Bigler has show me what I do wrong when trying to run the backend locally....

File: .gitignore [hunk 1]
```diff
@@ -16,7 +16,7 @@ target/
 !**/src/test/**/target/
 *.log
 create.sql
-update.sql
+update
```
Comment (Andy-Bigler) [inline .gitignore:19]: Why is this change needed?

File: src/main/kotlin/ch/bfh/habits/controllers/HabitController.kt [hunk 2]
```diff
@@ -9,6 +9,7 @@ import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.*
 
 @RestController
+@CrossOrigin(origins = ["http://localhost:3000"])
```
Comment (Andy-Bigler) [inline src/main/kotlin/ch/bfh/habits/controllers/HabitController.kt:12]: Thanks. Will configure it later for the whole app.

File: ui/.huskyrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "hooks": {
+        "pre-commit": "tsc && lint-staged"
```
Comment (Andy-Bigler) [inline ui/.huskyrc:3]: We can't use husky. As soon as you do this it will override the other pre commit hook in this repo. (So no hooks anymore for the backend) That's why I moved everything in there. Imo we can still merge this and clean up later.

File: update.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+alter table if exists habit add column description varchar(255);
```
Comment (Andy-Bigler) [inline update.sql:1]: This file does not need to be tracked.

**Review by Andy-Bigler — APPROVED** [review]

File: update.sql (hunk 4 above)
Comment (janlauber) [reply update.sql]: removed

File: ui/.huskyrc (hunk 3 above)
Comment (janlauber) [reply ui/.huskyrc]: removed

File: src/main/kotlin/ch/bfh/habits/controllers/HabitController.kt (hunk 2 above)
Comment (janlauber) [reply src/main/kotlin/ch/bfh/habits/controllers/HabitController.kt:12]: k

File: .gitignore (hunk 1 above)
Comment (janlauber) [reply .gitignore:18]: removed it
